docs/standard/garbage-collection/large-object-heap.md

```diff
@@ -0,0 +1,313 @@
+
+Since the LOH is only collected during generation 2 GCs, the LOH segment can only be freed during such a GC. Figure 3 illustrates a scenario where the garbage collector releases one segment (segment 2) back to the OS and decommits more space on the remaining segments. If it needs to use the decommitted space at the end of the segment to satisfy large object allocation requests, it commits the memory again. (For an explanation of commit/decommit, see the documentation for [VirtualAlloc](https://msdn.microsoft.com/library/windows/desktop/aa366887(v=vs.85).aspx).
+ 
+![Figure 3: LOH after a gen 2 GC](media/loh/loh-figure-3.jpg)  
+Figure 3: The LOH after a generation 2 GC
+
+## When is a large object collected?
+
+In general, a GC occurs when one of the following following 3 conditions happens:
+
+- Allocation exceeds the generation 0 or large object threshold.
+
+   The threshold is a property of a generation. A threshold for a generation is set when the garbage collector allocates objects into it. When the threshold is exceeded, a GC is triggered on that generation. When you allocate small or large objects, you consume generation 0 and the LOH’s thresholds, respectively. When the garbage collector allocates into generation 1 and 2, it consumes their thresholds. These thresholds are dynamically tuned as the program runs.
+
+   This is the typical case; most GCs happen because of allocations on the managed heap.
+
+- The <xref:System.GC.Collect%2A?displayProperty=nameWithType> method is called.
+
+   If the parameterless <xref:System.GC.Collect?displayProperty=nameWithType> method is called or another overload is passed <xref:System.GC.MaxGeneration?displayProperty=nameWithType> as an argument, the LOH is collected along with the rest of the managed heap.
+
+- The system is in low memory situation.
+
+   This occurs when the garbage collector receives a high memory notification from the OS. If the garbage collector thinks that doing a generation 2 GC will be productive, it triggers one.
+
+## LOH Performance Implications
+
+Allocations on the large object heap impact performance in the following ways.
+
+- Allocation cost.
+
+   The CLR guarantees that the memory for every new object it allocates will be cleared. This means the allocation cost of a large object is completely dominated by memory clearing (unless it triggers a GC). If it takes 2 cycles to clear one byte, it takes 170,000 cycles to clear the smallest large object. Clearing the memmory of a 16MB object on a 2GHz machine takes approximately 16ms. That's a rather large cost.
```

Comment:
please use the original wording here 'cause it's important - I had 
> The CLR makes the guarantee that the memory for every new object we give out is cleared

instead of what you have here

> The CLR guarantees that the memory for every new object it allocates will be cleared.

"will be cleared" does not specify when it will be cleared. we guarantee that when we hand out a new object, the memory is *already* cleared.